lib/oper.g

```diff
@@ -148,7 +148,26 @@ BIND_GLOBAL( "NUMBERS_PROPERTY_GETTERS", [] );
 ##  </ManSection>
 ##
 BIND_GLOBAL( "OPERATIONS", [] );
-
+BIND_GLOBAL( "OPER_FLAGS", rec() );
+BIND_GLOBAL( "STORE_OPER_FLAGS", 
+function(oper, flags)
+  ADD_LIST(OPERATIONS, oper);
+  # we need a back link to oper for the post-restore function
+  OPER_FLAGS.(HANDLE_OBJ(oper)) := [oper, flags];
+end);
+BIND_GLOBAL( "GET_OPER_FLAGS", oper-> OPER_FLAGS.(HANDLE_OBJ(oper))[2] );
+# the object handles change after loading a workspace
+ADD_LIST(GAPInfo.PostRestoreFuncs, function()
+  local tmp, a;
+  tmp := [];
+  for a in REC_NAMES(OPER_FLAGS) do
+    ADD_LIST(tmp, OPER_FLAGS.(a));
+    UNB_REC(OPER_FLAGS, a);
+  od;
+  for a in tmp do 
+    OPER_FLAGS.(HANDLE_OBJ(a[1])) := a;
+  od;
+end);
```

Comment:
I really like that this introduced wrappers `STORE_OPER_FLAGS` and `GET_OPER_FLAGS`, as that way, it's much easier to change the representation again later on.